File: 20220921/모범답안/20220921_10.js/car.js
/**
 * @param cars {array}
 * @param clients {array}
 * @return int
 */
function solution(cars, clients) {
    const x = 0
    const y = 1

    clients = clients.sort((a, b) => a[y] - b[y])
    cars = cars.sort((a, b) => a[x] - b[x])

    let sells = cars.reduce((total, car) => {
        for (let i = 0; i < clients.length; i++){
            const client = clients[i]
            if (client[y] >= car[y] && client[x] < car[x]) {
                clients.splice(i, 1)
                return total + 1
            }    
        }
        return total
    }, 0)

    return sells
}

File: 20220921/모범답안/20220921_10.js/car.py
def solution(cars, clients):
    '''
    :param cars: list
    :param clients: list
    :return: int
    '''
    answer = 0
    n = len(clients)
    cars.sort()
    bought_clients = [False] * n

    for car in cars:
        x, y = car
        eligible_clients = [(clients[i][1], i) for i in range(n) if (not bought_clients[i]) and clients[i][0] < x and clients[i][1] >= y]
        
        if len(eligible_clients):
            p, j = min(eligible_clients)
            bought_clients[j] = True
            answer += 1

    return answer
